snmp/datadog_checks/snmp/config.py

```diff
@@ -26,20 +27,32 @@ def __init__(self, instance, warning, global_metrics, mibs_path, profiles, profi
             self.metrics.extend(profiles[profile]['definition'])
         self.enforce_constraints = is_affirmative(instance.get('enforce_mib_constraints', True))
         self.snmp_engine, self.mib_view_controller = self.create_snmp_engine(mibs_path)
+        self.ip_address = None
+        self.network_address = None
+        self.discovered_instances = {}
 
         timeout = int(instance.get('timeout', self.DEFAULT_TIMEOUT))
         retries = int(instance.get('retries', self.DEFAULT_RETRIES))
-        self.transport = self.get_transport_target(instance, timeout, retries)
 
-        self.ip_address = instance['ip_address']
+        if 'ip_address' not in instance and 'network_address' not in instance:
+            raise ConfigurationError('An IP address or a network address needs to be specified')
+
+        if 'ip_address' in instance:
+            self.transport = self.get_transport_target(instance, timeout, retries)
+
+            self.ip_address = instance['ip_address']
+            self.tags.append('snmp_device:{}'.format(self.ip_address))
+
+        if 'network_address' in instance:
+            self.network_address = instance['network_address']
```

Comment:
The `InstanceConfig` probably need some unit tests to cover the existing and new configuration logics.

Reply:
Added a test for the error, the logic ought to be tested already.